resources/lambdas.md

```diff
@@ -78,23 +78,64 @@ For more on lambda expressions, refer to the
 This will be a basic introduction into functional programming, I recommend first reading the earlier section on lambdas and our section on [streams](/resources/streams) as it will help a lot and since 
 most of functional programming is streams and lambdas.
 
-## What is Functional Programming
+## What is a Functional Programming Language?
 
-Functional programming is a programming paradigm where functions are "treated as first class citizens". Essentially, functions are treated as objects and can do
-anything that an object can (be passed as a parameter, returned, etc).
-The main idea behind functional programming is that it's programming without any or with minimal [side effects](https://en.wikipedia.org/wiki/Side_effect_(computer_science)). 
-In other words, it's programming where only local variables that are created inside the function can be modified. This means that no global variables, variables passed by reference, or parameterized variables can be
-modified for it to count as functional programming. An example would be:
+In general programming theory, there exists two main types of languages:
```

Comment:
It's a bit more nuanced than this (as you touch on further downwards). Many languages have a ton of functional features while being imperative in nature. They support different paradigms - it's not a black and white case of "this language is either imperative or functional" (like this implies) and more a case of how your code specifically is written.

Maybe specify that these are paradigms, and can be used in conjunction (to make it more clear).